File: javascript/Array/reverseArr.js
function reverseArr(arr){
    let start = 0
    let end = arr.length-1
    while (start < end){
        const temp = arr[start]
        arr[start]= arr[end]
        arr[end] = temp
        start += 1
        end -= 1
    }
    return arr

}

console.log(reverseArr([1,2,3,4]))
console.log(reverseArr([1,0,3,4]))
console.log(reverseArr([7,7,7]))
console.log(reverseArr([7,20,30,10]))

File: javacode/src/main/java/Coding/ReverseArr.java
package Coding;

import java.util.Arrays;

public class ReverseArr {

    public static void main(String[] args) {
        int[] nums = {1,2,3,4,5,6,7,8};
        int[] reversed = getReversedArr(nums);
        System.out.println("reversed:: "+ Arrays.toString(reversed));
    }

    static int[]  getReversedArr(int[] nums) {

        int start = 0;
        int last = nums.length -1;
        int mid = nums.length / 2;
        System.out.println("mid:: "+mid);
        for (int i = 0; i < mid-1; i++) {
            int temp = nums[start];
            nums[start] =nums[last];
            nums[last] = temp;
            start++;
            last--;
        }
        return nums;
    }


}
